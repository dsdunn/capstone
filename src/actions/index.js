export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const updateUser = (user) => ({
  type: 'UPDATE_USER',
  user
})

export const setProfile = profile => ({
  type: 'SET_PROFILE',
  profile
})

export const addCollections = (collections) => ({
  type: 'ADD_COLLECTIONS',
  collections
})

export const setCollection = collection => ({
  type: 'SET_COLLECTION',
  collection
})

export const updateCollectionsList = (collectionsList) => {
  type: 'UPDATE_COLLECTIONS_LIST',
  collectionsList
}