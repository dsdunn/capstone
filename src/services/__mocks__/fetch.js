export const getAllCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollectionsByCategory = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserInfo = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666
    }])
  })

export const getSearchResults = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      collections:[]
    }])
  })

export const postCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'
    }])
  })