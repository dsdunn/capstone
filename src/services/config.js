export default {
  apiKey: '66642069',
  url: 'https://collecshare.herokuapp.com/api/v1'
}