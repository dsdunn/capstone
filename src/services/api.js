import config from './config';

export const apiFetch = (path, method = 'GET', payload = null) => {
  let body = payload ? JSON.stringify(payload) : null
  return fetch(`${config.url}${path}`, {
    method,
    headers: {
     'Content-Type': 'application/json',
     'Accept': 'application/json'
    },
    body
  })
}

export const apiFormDataFetch = (path, method = 'PUT', payload = null) => {
  console.log(path, method, payload);
  let body = payload; 
  return fetch(`${config.url}${path}`, {
    method,
    body
  })
}
