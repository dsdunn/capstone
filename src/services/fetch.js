import { apiFetch, apiFormDataFetch } from './api'

export const postUserInfo = user => {
  return apiFetch('/users', 'POST', user)
}

export const getUserInfo = uid => {
  return apiFetch(`/users/${uid}`, 'GET')
  .then(response => response.json())
}

export const putUserInfo = (userInfo, uid) => {
  return apiFormDataFetch(`/users/${uid}`, 'PUT', userInfo)
  .then(response => console.log(response))
  .then(response => response.json())
}

export const postCollection = collection => {
  return apiFetch(`/collections`, `POST`, collection)
  .then(response => response.json())
}

export const getAllCollections = () => {
  return apiFetch('/collections')
  .then(response => response.json())
}

export const getUserCollections = uid => {
  return apiFetch(`/collections?uid=${uid}`)
  .then(response => response.json())
}