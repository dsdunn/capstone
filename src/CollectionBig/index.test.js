import React from 'react';
import { CollectionBig } from './';
import { shallow } from 'enzyme';
import { getUserInfo } from '../services/fetch';
import { mapStateToProps, mapDispatchToProps } from './index.js'
import { setProfile } from '../actions';

jest.mock('../services/fetch');

describe('CollectionBig', () => {
  let wrapper;
  let mockCollection = {
    location: 'yo mama',
    id: 1,
    uid: 1,
    username: 'yo mama',
    title: 'yo mama',
    description: 'so fat',
    avatar: 'yo mama',
    image: 'yo mama',
    items: ['this', 'that']
  }

  beforeEach(() => {
    wrapper = shallow(<CollectionBig 
      history={{location: 'yo mamas', push: jest.fn()}} 
      collection={mockCollection } 
      setProfile={jest.fn()}/>); 
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call goToProfile when avatar is clicked', () => {
    wrapper.find('.collection-big-avatar').simulate('click');
    expect(getUserInfo).toBeCalled();
  })

  it('should return a props object with setProfile', () => {
    const mockDispatch = jest.fn();
    const mockProfile = { password1: '123456',
                       password2: '123456',
                       email: 'heck@yeah.com',
                       username: 'Sparkle' }
                       

    const mappedProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = setProfile(mockProfile);

    mappedProps.setProfile(mockProfile);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should return a props object', () => {
    const mockState = {collection: "this and that"};
    const expected = {collection: "this and that"};
    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })

})


