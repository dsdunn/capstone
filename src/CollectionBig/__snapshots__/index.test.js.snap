// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionBig should match the snapShot 1`] = `undefined`;
